usageprocessor/processor.go

```diff
@@ -2,33 +2,55 @@ package usageprocessor
 
 import (
 	"context"
+	"fmt"
 
+	hnyext "github.com/honeycombio/opentelemetry-collector-configs/honeycombextension"
+	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/pdata/plog"
 	"go.opentelemetry.io/collector/pdata/pmetric"
 	"go.opentelemetry.io/collector/pdata/ptrace"
 )
 
+var (
+	unset component.ID
+)
+
 type usageProcessor struct {
+	config   *Config
+	recorder hnyext.HoneycombUsageRecorder
 }
 
 func newUsageProcessor() (*usageProcessor, error) {
 	return &usageProcessor{}, nil
 }
 
+func (p *usageProcessor) Start(ctx context.Context, host component.Host) error {
+	if p.config.honeycombExtensionID != unset {
+		ext := host.GetExtensions()[p.config.honeycombExtensionID]
+		if ext == nil {
+			return fmt.Errorf("extension %q does not exist", p.config.honeycombExtensionID.String())
+		}
+
+		recorder, ok := ext.(hnyext.HoneycombUsageRecorder)
+		if !ok {
+			return fmt.Errorf("extension %q does not implement HoneycombUsageRecorder", p.config.honeycombExtensionID.String())
+		}
+		p.recorder = recorder
+	}
+	return nil
+}
+
 func (p *usageProcessor) processTraces(ctx context.Context, tracesData ptrace.Traces) (ptrace.Traces, error) {
-	// m := ptrace.ProtoMarshaler{}
-	// size := m.TracesSize(tracesData)
+	p.recorder.RecordTracesUsage(tracesData)
```

Comment:
I feel like the processor should handle calculating the value and the recorder should only be used to record so that the processor is in full control of the "math" used to calculate the datapoint. I cant give a strong argument for why I feel this way tho.

Reply:
I like that the recorder takes the whole data stuct (traces, metrics, logs) as we could expands the number of things we calculate from the data later (eg number of spans, metrics, log records, etc) without making the processor do all the work itself.

Reply:
A benefit of the processor doing the work is that we'd be more likely to be able to upstream it. If we do any math in the processor and let the recording of the data be generic then we decouple the data generation and recording aspects.